mapbox/libjava-services/src/main/java/com/mapbox/services/api/utils/turf/TurfMeta.java

```diff
@@ -0,0 +1,108 @@
+import com.mapbox.services.commons.geojson.MultiPolygon;
+import com.mapbox.services.commons.geojson.Point;
+import com.mapbox.services.commons.geojson.Polygon;
+import com.mapbox.services.commons.models.Position;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TurfMeta {
+
+  /**
+   * Get all coordinates from any GeoJSON object, returning a {@code List} of Position objects.
+   *
+   * @param layer            Any GeoJSON object.
+   * @param excludeWrapCoord whether or not to include the final coordinate of LinearRings that wraps the ring in its
+   *                         iteration.
+   * @return A {@code List} made up of {@link Position}s.
+   * @throws TurfException Occurs when the GeoJSON provided isn't valid.
+   * @since 2.0.0
+   */
+  public static List<Position> coordEach(GeoJSON layer, boolean excludeWrapCoord) throws TurfException {
+    int i;
+    int g;
+    int j;
+    int k;
+    int l;
+    int stopG;
+    boolean isGeometryCollection;
+    GeoJSON geometryMaybeCollection;
+    Geometry geometry;
+    List<Position> coords = new ArrayList<>();
+    boolean isFeatureCollection = layer.getType().equals("FeatureCollection");
+    boolean isFeature = layer.getType().equals("Feature");
+    int stop = isFeatureCollection ? ((FeatureCollection) layer).getFeatures().size() : 1;
+
+    // This logic may look a little weird. The reason why it is that way is because it's trying to be fast. GeoJSON
+    // supports multiple kinds of objects at its root: FeatureCollection, Features, Geometries. This function has the
+    // responsibility of handling all of them, and that means that some of the `for` loops you see below actually just
+    // don't apply to certain inputs. For instance, if you give this just a Point geometry, then both loops are
+    // short-circuited and all we do is gradually rename the input until it's called 'geometry'.
+    //
+    // This also aims to allocate as few resources as possible: just a few numbers and booleans, rather than any
+    // temporary arrays as would be required with the normalization approach.
+    for (i = 0; i < stop; i++) {
+
+      geometryMaybeCollection = (isFeatureCollection ? ((FeatureCollection) layer).getFeatures().get(i).getGeometry() :
+        (isFeature ? ((Feature) layer).getGeometry() : layer));
+
+      isGeometryCollection = geometryMaybeCollection.getType().equals("GeometryCollection");
+      stopG = isGeometryCollection ? ((GeometryCollection) geometryMaybeCollection).getGeometries().size() : 1;
+
+      for (g = 0; g < stopG; g++) {
+
+        geometry = isGeometryCollection
+          ? (((GeometryCollection) geometryMaybeCollection).getGeometries().get(g))
+          : (Geometry) geometryMaybeCollection;
+
+        int wrapShrink = (excludeWrapCoord
+          && (geometry.getType().equals("Polygon") || geometry.getType().equals("MultiPolygon")) ? 1 : 0);
+
+        if (geometry.getType().equals("Point")) {
```

Comment:
I'm not completely satisfied with this approach as it goes around Java's type system. How about instead of having one only `TurfMeta.coordEach(GeoJSON layer, boolean excludeWrapCoord)` that takes a generic `GeoJSON` object, we have different `TurfMeta.coordEach()` methods for each supported GeoJSON type? Something like [what we do here](https://github.com/mapbox/mapbox-java/blob/master/mapbox/libjava-services/src/main/java/com/mapbox/services/api/utils/turf/TurfMeasurement.java) with `destination()`.